artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/AddressSettings.java

```diff
@@ -1835,6 +1856,8 @@ public void encode(ActiveMQBuffer buffer) {
       BufferHelper.writeNullableInteger(buffer, prefetchPageBytes);
 
       BufferHelper.writeNullableInteger(buffer, prefetchPageMessages);
+
+      BufferHelper.writeNullableBoolean(buffer, autoCreateDivertDestination);
```

Comment:
don't think we should change these.

Reply:
new version now is using the JSON encoded on the journal.. the older stuff is just for the journals... so these are not needed I think.

Reply:
@clebertsuconic Thanks, I have made the changes you suggested. The only thing I'm noticing now is that if I inject this setting in some current persistence tests they start failing, though I guess that might be expected with this new method?

One example of a test that would fail is for instance:
`org.apache.activemq.artemis.tests.integration.persistence.AddressSettingsConfigurationStorageTest#testStoreSecuritySettings()` where if I add some random previous address setting (like `setAutoDeleteAddresses(true)`) the test still passes, whereas if I add `setAutoCreateDivertDestination(true)` the test fails.

Is that right/expected or am I missing something?

Reply:
I will run the tests and take a look... will comment back here after my findings by the end of my day .

Reply:
@AntonRoskvist Can you add your change and rebase your PR on top of https://github.com/apache/activemq-artemis/pull/4854


The tests were still using the old method. I am removing *any* usage from the AddressSettings usage, and I am now making it *clearer* with a big javadoc to not have any more additions.


The reason that you needed to add a field now was a reason I added the JSON format. I got tired to see that method's wire being changed over time.  That got even worse when I played with my private branches when I was only bringing the newer field and not the previous field.

So, if I added field A and B on apache/master

and only Brough field B to a private branch, my wire would be inconsistent with apache/main.  It was being clusterFreak to manage it :)


If you could bring my commit to your PR (Rebase as I said), and bring any changes you like, it should now work.

Reply:
@AntonRoskvist I have merged #4854... if you rebase against main, you can add anything to the tests you mentioned and it should work now.

I have removed any usage of the older encoder.. just keeping the decoder for compatibility with older journals.

Reply:
@clebertsuconic Yes, looking good after rebasing. Thanks!